### ISSUE TITLE
unescape_string is missing support for octal escape sequences

### ISSUE THREAD

**Issue by toffaletti** [nushell/nushell#5551]
### Describe the bug

`printf "\033]133;A\007"` this was working with 0.61 but is no longer working with 0.62

```
printf "\033]133;A\007" 
Error: nu::parser::parse_mismatch (link)

  × Parse mismatch during operation.
   ╭─[entry #1:1:1]
 1 │ printf "\033]133;A\007"
   ·                   ──┬──
   ·                     ╰── expected supported escape character
   ╰────
```

### How to reproduce

```
printf "\033]133;A\007" 
Error: nu::parser::parse_mismatch (link)

  × Parse mismatch during operation.
   ╭─[entry #1:1:1]
 1 │ printf "\033]133;A\007"
   ·                   ──┬──
   ·                     ╰── expected supported escape character
   ╰────
```

### Expected behavior

I expected this to continue working as octal are often used for terminal escape sequences.

### Configuration

| key                | value                                           |
| ------------------ | ----------------------------------------------- |
| version            | 0.62.0                                          |
| branch             |                                                 |
| tag                |                                                 |
| build_os           | macos-x86_64                                    |
| rust_version       | rustc 1.60.0 (7737e0b5c 2022-04-04)             |
| rust_channel       | stable-x86_64-apple-darwin                      |
| cargo_version      | cargo 1.60.0 (d1fd9fe2c 2022-03-01)             |
| pkg_version        | 0.62.0                                          |
| build_time         | 2022-05-15 22:25:54 -07:00                      |
| build_rust_channel | release                                         |
| features           | database, dataframe, default, trash, which, zip |
| installed_plugins  |                                                 |

**Comment by fdncred** [comment]
good catch. we'd accept a pr to fix this.

**Comment by hustcer** [comment]
@toffaletti Does this issue has been fixed? BTW: there is no `printf` command, is it a alias ?

**Comment by toffaletti** [comment]
Now that I've spent some time understanding what is happening here, I do not think the issue has been addressed. The way I see the problem now is that printf is an external command and nu is applying its own rules to the arguments of the external command, this is similar to the go test ./... issue or jq . issues previously. There are some trade offs either way, but some pragmatic decision needs to be made about how to pass arguments to externals that would otherwise be interpreted by nu to mean something.

**Comment by sholderbach** [comment]
@toffaletti agreed we should support that either raw escaped strings (maybe with slightly special syntax) can be passed to down stream external tools or that we locally translate expected escape codes and pass the corresponding string. (Do we have to support `OsString`s for that? Or is valid UTF-8 ok (Windows is probably itself somewhat more restrictive when it comes to valid strings but internally UTF-16 encoded))

**Comment by Ayanrocks** [comment]
Can this be taken up? Any help on getting started. A new rust dev here.

**Comment by weirdan** [comment]
I believe this was addressed with the introduction of raw strings:

```console
❯ printf r#'\033]133;A\007'# | into binary
Length: unknown (stream) | printable whitespace ascii_other non_ascii
00000000:   1b 5d 31 33  33 3b 41 07                             •]133;A•
```
